### PR TITLE
324 pp check multinomial

### LINKED ISSUE #324: [feature-request] Add `pp_check()` support for multinomial models (`m3`)
## Problem

`brms::pp_check()` does not support the multinomial family, so calling it on a fitted m3 model fails. The m3 vignette documents a manual workaround using `tidybayes::epred_draws()`, but it would be nice for `pp_check(fit)` to work for all models.

## Planned fix
Add a `pp_check.bmmfit()` S3 method that intercepts multinomial models and dispatches to a custom visualization. For non-multinomial models, it delegates to `brms::pp_check()` via `NextMethod()`.

The multinomial handler:

Calls `brms::posterior_predict()` (already works for `m3`)
Pools observed and predicted counts by experimental condition
Normalizes to proportions per response category
Plots predicted mean + credible band vs observed proportions, faceted by condition

## Changes

- New file `R/pp_check.R` with `pp_check.bmmfit()` and internal helpers
- Update `m3` vignette to use `pp_check()` as primary approach (keep tidybayes as advanced alternative)
- New test file `tests/testthat/test-pp_check.R`

This also lays the groundwork for other models that rely on categorical or multinomial distributions such as the SDT rating model.

### PR DESCRIPTION
**Pull request by GidonFrischkorn**
## Summary

- Add `pp_check()` S3 method for `bmmfit` objects that handles multinomial models (e.g., m3) where `brms::pp_check()` is unavailable
- For multinomial models, produces a `ppc_bars`-style plot: observed proportions as bars, posterior predictive medians with credible intervals as point-ranges, using bayesplot's default colour scheme and theme
- For all other model families, delegates to `brms::pp_check()` via `NextMethod()`, with automatic `_grouped` type selection when `group` is specified
- Supports `group` faceting, configurable credible intervals (`probs`), and `re_formula` pass-through for population-level predictions
- Updates the m3 vignette to use `pp_check()` as the primary model-checking approach
- Adds `bayesplot` and `rlang` to DESCRIPTION Imports — both are already indirect dependencies via `brms` and add no installation burden

## Test plan

- [x] 32 new tests covering multinomial plots, group faceting, auto-grouped type selection, brms delegation, `re_formula`, and edge cases
- [x] `devtools::test(filter = "pp_check")` passes (0 failures)
- [x] `R CMD check` passes with 0 errors, 0 warnings

### CODE REVIEW

**Comment by GidonFrischkorn** [comment]
I am looking into the failing checks. I tried to find a leightweight option to perform proper checks for the pp_check implementation and locally the R CMD check did not complain.

Will upload a fix asap.

File: R/pp_check.R [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+    return(grouped)
+  }
+  type
+}
+
+
+.pp_check_multinomial <- function(object, type = NULL, ndraws = NULL,
+                                  probs = c(0.025, 0.975),
+                                  group = NULL,
+                                  # brms pp_check args — intercepted, not forwarded
```
[the remark this answers is not in the document]
Comment (GidonFrischkorn) [reply R/pp_check.R]: probs is an opt-in argument passed through ... by users who intentionally override the default. If malformed, stats::quantile() already produces clear error messages (e.g., "probs outside [0,1]"). Adding a separate validation layer would duplicate what base R already handles. This is consistent with how brms treats probs in its own pp_check — no upstream validation, the downstream function handles it.

File: R/pp_check.R [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+#'
+#' For models where brms provides `pp_check` support, this method delegates to
+#' [brms::pp_check()]. For models with multinomial families (e.g., the m3
+#' model), brms's `pp_check` is unavailable; this method dispatches to a
+#' model-specific visualisation instead.
+#'
+#' For **multinomial models**, the plot mirrors the bayesplot `ppc_bars` style:
+#' observed proportions are shown as bars and posterior predictive medians with
+#' credible intervals are shown as point-ranges, using the bayesplot default
+#' colour scheme and theme.
```
[the remark this answers is not in the document]
Comment (GidonFrischkorn) [reply R/pp_check.R:16]: The hex values used (#d1e1ec, #b3cde0, #03396c) are the actual colours bayesplot uses internally in ppc_bars. They are hard-coded here because bayesplot does not export a palette accessor function. The wording is accurate — these are the bayesplot defaults — but we could tighten it to "colours matching the bayesplot ppc_bars style" if preferred.

File: R/pp_check.R [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+
+
+.pp_check_multinomial <- function(object, type = NULL, ndraws = NULL,
+                                  probs = c(0.025, 0.975),
+                                  group = NULL,
+                                  # brms pp_check args — intercepted, not forwarded
+                                  x = NULL, prefix = c("ppc", "ppd"),
+                                  draw_ids = NULL,
+                                  nsamples = NULL, subset = NULL,
+                                  ...) {
```
[the remark this answers is not in the document]
Comment (GidonFrischkorn) [reply R/pp_check.R]: This is intentional. The parameters x, prefix, draw_ids, nsamples, and subset are explicitly listed in the signature to intercept them — they are standard brms::pp_check() arguments that users might pass out of habit but that don't apply to the multinomial visualisation. Without interception, they would be forwarded to posterior_predict() via ... and cause errors. The comment on line 69 documents this: "brms pp_check args — intercepted, not forwarded."

File: tests/testthat/test-pp_check.R [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    pp_check(fit, type = "hist", ndraws = 5),
+    "type.*ignored"
+  )
+})
+
+test_that("pp_check() uses bayesplot theme_default", {
+  skip_if_not_installed("ggplot2")
+  fit <- load_m3_fit()
+  p <- pp_check(fit, ndraws = 5)
+  expect_equal(p$theme$text$family, "serif")
```
[the remark this answers is not in the document]
Comment (GidonFrischkorn) [reply tests/testthat/test-pp_check.R:82]: The family == "serif" assertion is a deliberate smoke test confirming that bayesplot::theme_default() was applied (serif is its documented default font). If bayesplot changes this in a future release, a test failure would be the correct signal — we would want to review whether our styling still aligns. The test is intentionally specific rather than permissive.

**Comment by GidonFrischkorn** [comment]
I am not entirely sure, if I added the internal files for the tests correctly. But they do run now and have also been reduced in size to keep the package size small.

File: R/pp_check.R [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+    warning2("Argument 'type' is ignored for multinomial pp_check. ",
+             "A response-proportion plot is always produced.")
+  }
+
+  if (!is.null(nsamples)) {
+    warning2("'nsamples' is deprecated; use 'ndraws' instead.")
+    if (is.null(draw_ids)) ndraws <- nsamples
+  }
+
+  if (is.null(ndraws)) ndraws <- 100L
```
Comment (venpopov) [inline R/pp_check.R]: instead of an in-body check for NULL, make 100L the default argument, which makes it transparent to users what the default is

File: R/pp_check.R [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+
+  if (!is.null(type) && type != "dens_overlay") {
+    warning2("Argument 'type' is ignored for multinomial pp_check. ",
+             "A response-proportion plot is always produced.")
+  }
+
+  if (!is.null(nsamples)) {
+    warning2("'nsamples' is deprecated; use 'ndraws' instead.")
+    if (is.null(draw_ids)) ndraws <- nsamples
+  }
```
Comment (venpopov) [inline R/pp_check.R]: no need for this check since posterior_predict checks it, when you simplify as I explain next

File: R/pp_check.R [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+  predict_args <- if (!is.null(draw_ids)) {
+    list(draw_ids = draw_ids)
+  } else {
+    list(ndraws = ndraws)
+  }
+
+  model <- object$bmm$model
+  resp_cols <- unlist(model$resp_vars)
+  yrep <- do.call(brms::posterior_predict,
+                  c(list(object), predict_args, list(...)))
```
Comment (venpopov) [inline R/pp_check.R]: This can be simplified to 

```r
brms::posterior_predict(object, draw_ids = draw_ids, ndraws = ndraws, ...)
```

posterior_predict description already says it determines whether to use draw_ids or ndraws

File: R/pp_check.R [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+                                  probs = c(0.025, 0.975),
+                                  group = NULL,
+                                  # brms pp_check args — intercepted, not forwarded
+                                  x = NULL, prefix = c("ppc", "ppd"),
+                                  draw_ids = NULL,
+                                  nsamples = NULL, subset = NULL,
+                                  ...) {
+  if (!requireNamespace("ggplot2", quietly = TRUE)) {
+    stop2("ggplot2 is required for pp_check of multinomial models.")
+  }
```
Comment (venpopov) [inline R/pp_check.R]: not needed since we import brms, which imports bayesplot, which imports ggplot2. ggplot2 is a hard dependency whether we have it explicitly listed or not

File: R/pp_check.R (hunk 3 above)
[the remark this answers is not in the document]
Comment (venpopov) [reply R/pp_check.R]: Could you explain why that's necessary? subset is a deprecate argument; draw_ids and nsamples can be handled better as I explain below (unless I am overlooking something, which is always possible). Rather than us proactively intercepting arguments, I'd rather let the error happen and not have to maintain extra code  (and not silently hide that this arguments are ignored!). Alternatively, to make this more transparent, do this in the begining of the function:

```r
dots <- list(...)
ignored <- c("x", "prefix", "subset")
## if ignored in names(dots) then warn the user that arguments are ignored
```

**Review by venpopov — CHANGES_REQUESTED** [review]

**Comment by venpopov** [comment]
The failing tests are weird. There is no "update with develop" button as usual, so why are the files for the emmeans test missing from this branch? Not a question for you, but I'm confused as to what github is doing here

**Review by venpopov — APPROVED** [review]

**Comment by venpopov** [comment]
Oh, I see. You moved them to the assets folder here in a separate commit, but in the emmeans PR and now in dev they were in extdata and the test still reflect that. I will fix it
